pkg/compactor/compactor_metrics.go

```diff
@@ -0,0 +1,199 @@
+package compactor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/compact"
+	"github.com/thanos-io/thanos/pkg/extprom"
+)
+
+type compactorMetrics struct {
+	reg              prometheus.Registerer
+	commonLabels     []string
+	compactionLabels []string
+
+	// block.BaseFetcherMetrics
+	baseFetcherSyncs *prometheus.CounterVec
+
+	// block.FetcherMetrics
+	metaFetcherSyncs        *prometheus.CounterVec
+	metaFetcherSyncFailures *prometheus.CounterVec
+	metaFetcherSyncDuration *prometheus.HistogramVec
+	metaFetcherSynced       *extprom.TxGaugeVec
+	metaFetcherModified     *extprom.TxGaugeVec
+
+	// compact.SyncerMetrics
+	syncerGarbageCollectedBlocks    *prometheus.CounterVec
+	syncerGarbageCollections        *prometheus.CounterVec
+	syncerGarbageCollectionFailures *prometheus.CounterVec
+	syncerGarbageCollectionDuration *prometheus.HistogramVec
+	syncerBlocksMarkedForDeletion   *prometheus.CounterVec
+
+	compactions                 *prometheus.CounterVec
+	compactionPlanned           *prometheus.CounterVec
+	compactionRunsStarted       *prometheus.CounterVec
+	compactionRunsCompleted     *prometheus.CounterVec
+	compactionFailures          *prometheus.CounterVec
+	verticalCompactions         *prometheus.CounterVec
+	remainingPlannedCompactions *prometheus.GaugeVec
+}
+
+const (
+	UserLabelName      = "user"
+	TimeRangeLabelName = "time_range_milliseconds"
+	ReasonLabelName    = "reason"
+)
+
+var (
+	CommonLabels     = []string{UserLabelName}
+	CompactionLabels = []string{TimeRangeLabelName}
+)
```

Comment:
Nits. Can we keep those variables private? As they are not used by other packages.